pages/coinbase/homepage.py

```diff
@@ -1,27 +1,37 @@
 
 import config
 from pages.templates.elements import *
-
+from selenium.common.exceptions import NoSuchElementException
 
 
 class HomePage:
 
-    def __init__(self, driver, spawn=True):
+    def __init__(self, driver, spawn=True, close_cookie_banner=True):
         """
         inputs
         -----
         driver = selenium driver
         spawn: bool
-            The default use case is that the driver being passed in is new and 
+            The default use case is that the driver being passed in is new and
             needs to be navigated to the homepage. if spawn=False, it is assumed
             that the driver passed in is already at the homepage.
         """
-
         driver.get(config.coinbase_domain)
-        self.driver = driver
 
+        self.driver = driver
         self.build_elements()
 
+        if close_cookie_banner:
+            wait(self.driver,10).until(EC.visibility_of_element_located(
+                self.visible_cookie_banner_loc))
+            self.cookie_banner_dismiss_button.click()
+
+
+    # A dismiss cookie dialogue box appears and hides elements, clicking
+    # the dismiss button performed in the init method fixes this issue  
+    visible_cookie_banner_loc = (by.XPATH, "//div[@class='sc-fzoLsD gkNzOD']")
```

Comment:
This isn't the xpath for the banner, it's the xpath for the text on the banner. Rename this (to something like "banner_text_loc") or put in the xpath for the actual banner. I prefer the former, but if you run into problems don't worry about it too much.

Also don't use class names if they're random-seeming strings. These sometimes change randomly. Use the contained text instead ("Cookie Policy" should work fine, then you can work up to the banner element from there).

- [ ] remove the 'visible_' part of this (unless there's a good reason for it I'm not thinking of?)
- [ ] rename the locator or use the xpath for the banner
- [ ] don't use the class name to find the element